### ISSUE TITLE
Logs in journal: QDBusObjectPath: invalid path ""

### ISSUE THREAD

**Issue by stanczew** [CDrummond/cantata#1082]
OS: Arch Linux with KDE Plasma 5.10.5
Cantata version: 2.1.0 (also checked with current git, 96fb1dea), compiled with `-DENABLE_LIBVLC=OFF`
MPD version: 0.20.10

Hi,

I noticed my journal gets filled with the following messages:
`plasmashell[526]: QDBusObjectPath: invalid path ""`
About every 2 seconds, two copies of the message appear. This only happens when a) Cantata is running and b) a song is currently playing. After stopping playback or closing Cantata the messages no longer appear.

I have narrowed the issue down to function `Mpris::signalUpdate` in `dbus/mpris.cpp`. About every 2 seconds, a signal for property Position gets emitted. The last element of `args` list is an empty QStringList; removing it from `args` makes the issue go away (logs don't appear in journal anymore).
However, I'm not sure if this is a correct fix, or if there is a reason for this empty QStringList to be there, so I wanted to open an issue to sort it out.

Also, I digged a little bit through MPRIS specification [1], and found the following statement about the Position property:
> The `org.freedesktop.DBus.Properties.PropertiesChanged` signal is **not** emitted when this property changes.

From what I understand, subscribers should be able to calculate Position themselves using the Rate property, only reading Position after Seeked signal is emitted (e.g. when the user clicks on the progress bar). I see that Cantata emits PropertiesChanged signal with new Position value instead of Seeked signal; is this behavior expected? Again, I don't know much about this – just wanted to point out a potential problem.

Thank you,
— Maciej

[1] https://specifications.freedesktop.org/mpris-spec/latest/Player_Interface.html#Property:Position

**Comment by CDrummond** [comment]
Sorry I dont see any messages in journal. `journalctl  --follow` ??

The last arg is required for PropertiesChanged, otherwise the signal does not work. It's the invalidated properties - but Cantata sends the new values, so this is not used. Checked and Clementine is the same.

Not sure what harm emitting the signal more often does.

**Comment by stanczew** [comment]
Ok, I have looked further into Plasma and found a bug report:
https://bugs.kde.org/show_bug.cgi?id=383557

So, Cantata sends a Metadata property with empty contents, and Plasma tries to extract `mpris:trackid` from it, resulting in an error that gets logged. When I removed Metadata from the map, the logs did not appear.

I also checked Clementine, and I did *not* encounter the issue. Like you said, Clementine sends this signal in the same way, so it probably sets Metadata differently.

Anyway, I have built plasma-workspace with attached patch, and the issue is fixed.
Thanks for your time. Closing.
